File: api/app/Models/Product.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\SoftDeletes;

class Product extends Model
{
    use HasFactory, SoftDeletes;

    protected $fillable = [
        'sku',
        'name',
        'description',
        'price',
        'wholesale_price',
        'original_price',
        'unit',
        'quantity_per_unit',
        'image',
        'category_id',
        'agent_id',
        'is_available',
    ];

    protected $casts = [
        'price' => 'decimal:2',
        'wholesale_price' => 'decimal:2',
        'original_price' => 'decimal:2',
        'quantity_per_unit' => 'decimal:2',
        'is_available' => 'boolean',
    ];

    public function category()
    {
        return $this->belongsTo(Category::class);
    }

    public function agent()
    {
        return $this->belongsTo(User::class, 'agent_id');
    }

    public function orderItems()
    {
        return $this->hasMany(OrderItem::class);
    }
}

File: mobile/lib/models/product.dart
class Product {
  final int id;
  final String name;
  final String? description;
  final double price;
  final String? image;
  final int categoryId;
  final bool isAvailable;

  Product({
    required this.id,
    required this.name,
    this.description,
    required this.price,
    this.image,
    required this.categoryId,
    required this.isAvailable,
  });

  factory Product.fromJson(Map<String, dynamic> json) {
    return Product(
      id: json['id'],
      name: json['name'],
      description: json['description'],
      price: double.parse(json['price'].toString()),
      image: json['image'],
      categoryId: json['category_id'],
      isAvailable: json['is_available'] ?? true,
    );
  }
}





